test/vars/MavenStagingHelperSpec.groovy

```diff
@@ -0,0 +1,98 @@
+        1 * getPipelineMock("sh")([script: 'find FOLDER -name *.properties', returnStdout: true]) >> { return 'file.properties' }
+        1 * getPipelineMock("readProperties")([file: 'file.properties']) >> { return ['stagingRepository.id':'STAGING_ID'] }
+        'STAGING_ID' == helper.stagingRepositoryId
+    }
+
+    def "[MavenStagingHelper.groovy] stageLocalArtifacts empty stagingProfileId"() {
+        setup:
+        def helper = new MavenStagingHelper(steps)
+        when:
+        helper.stageLocalArtifacts('', 'FOLDER')
+        then:
+        thrown(AssertionError)
+    }
+
+    def "[MavenStagingHelper.groovy] stageLocalArtifacts empty artifacts folder"() {
+        setup:
+        def helper = new MavenStagingHelper(steps)
+        when:
+        helper.stageLocalArtifacts('ID', '')
+        then:
+        thrown(AssertionError)
+    }
+
+    def "[MavenStagingHelper.groovy] promoteStagingRepository"() {
+        setup:
+        def helper = new MavenStagingHelper(steps)
+        when:
+        helper.withStagingRepositoryId('STAGING_ID')
+            .promoteStagingRepository('ID')
+        then:
+        1 * getPipelineMock("sh")([script: 'mvn -B -q help:evaluate -Dexpression=project.artifactId -DforceStdout', returnStdout: true]) >> { return 'NAME' }
+        1 * getPipelineMock("sh")([script: 'mvn -B -q help:evaluate -Dexpression=project.version -DforceStdout', returnStdout: true]) >> { return 'VS' }
+        1 * getPipelineMock("sh")([script: "mvn -B --projects :NAME org.sonatype.plugins:nexus-staging-maven-plugin:1.6.5:promote -DnexusUrl=https://repository.jboss.org/nexus -DserverId=jboss-releases-repository -DstagingDescription='NAME VS' -DbuildPromotionProfileId=ID -DstagingRepositoryId=STAGING_ID", returnStdout: false])
+    }
+
+    def "[MavenStagingHelper.groovy] promoteStagingRepository empty buildPromotionProfileId"() {
+        setup:
+        def helper = new MavenStagingHelper(steps)
+        when:
+        helper.withStagingRepositoryId('STAGING_ID')
+            .promoteStagingRepository('')
+        then:
+        thrown(AssertionError)
+    }
+
+    def "[MavenStagingHelper.groovy] promoteStagingRepository no stagingRepositoryId"() {
+        setup:
+        def helper = new MavenStagingHelper(steps)
+        when:
+        helper.promoteStagingRepository('ID')
+        then:
+        thrown(AssertionError)
+    }
+
+    def "[MavenStagingHelper.groovy] full process"() {
+        setup:
+        def helper = new MavenStagingHelper(steps)
+        when:
+        helper.withNexusReleaseUrl('NEXUS_URL')
+            .withNexusReleaseRepositoryId('NEXUS_REPOSITORY_ID')
+            .withStagingDescription('DESCRIPTION')
+        helper.stageLocalArtifacts('STAGE_PROFILE_ID', 'FOLDER')
+        helper.promoteStagingRepository('BUILD_PROMOTE_ID')
```

Comment:
this is so theoretical, but shouldn't it be
```
setup:
        def helper = new MavenStagingHelper(steps)
        helper.withNexusReleaseUrl('NEXUS_URL')
            .withNexusReleaseRepositoryId('NEXUS_REPOSITORY_ID')
            .withStagingDescription('DESCRIPTION')
        helper.stageLocalArtifacts('STAGE_PROFILE_ID', 'FOLDER')
        when:
        helper.promoteStagingRepository('BUILD_PROMOTE_ID')
```
because you are arranging the test (helper stuff execution before promoting) to execute `promoteStagingRepository`

don't take me wrong, just proposing, it is weird for me to have more than one execution in the `Act` (`when`) section

Reply:
seems fair :)